proto/buf/alpha/registry/v1alpha1/repository_track_commit.proto

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+// RepositoryTrackCommit is the existance of a RepositoryCommit on a RepositoryTrack. Currently its only purpose is
+// for querying whether a RepositoryCommit is on a RepositoryTrack and determining it's sequence id.
+message RepositoryTrackCommit {
+  // We reserve field number '3' for id, but RepositoryTrackCommits have no id.
+  // string id = 1;
+
+  // immutable
+  google.protobuf.Timestamp create_time = 2;
+
+  // We reserve field number '3' for the update_time.
+  // google.protobuf.Timestamp update_time = 3;
+
+  // immutable
+  string repository_track_id = 4;
+
+  // immutable
+  string repository_commit_id = 5;
+
+  // unique for repository_track, immutable
+  int64 sequence_id = 6;
+}
+
+service RepositoryTrackCommitService {
+  // GetRepositoryTrackCommitByCommitReference returns the RepositoryTrackCommit associated with the given
+  // CommitReference on the given RepositoryTrack. Returns NOT_FOUND if the RepositoryTrackCommit does not exist.
+  rpc GetRepositoryTrackCommitByCommitReference(GetRepositoryTrackCommitByCommitReferenceRequest) returns (GetRepositoryTrackCommitByCommitReferenceResponse);
+  // ListRepositoryTrackCommitsByRepositoryTrack lists the commits associated with a repository track, ordered
+  // by their sequence id.
+  rpc ListRepositoryTrackCommitsByRepositoryTrack(ListRepositoryTrackCommitsByRepositoryTrackRequest) returns (ListRepositoryTrackCommitsByRepositoryTrackResponse);
+}
+
+message GetRepositoryTrackCommitByCommitReferenceRequest {
+  // The id of the repository containing the commit.
+  string repository_id = 1;
+  // The name of the repository track.
+  string repository_track_name = 2;
+  // The commit reference to resolve.
+  string commit = 3;
+}
+
+message GetRepositoryTrackCommitByCommitReferenceResponse {
+  RepositoryTrackCommit repository_track_commit = 1;
+}
+
+message ListRepositoryTrackCommitsByRepositoryTrackRequest {
```

Comment:
I think we normally keep it consistent for IDs with other IDs, and names with other names. Should we make this act upon `repository_id` and `repository_track_id`? Then we could reserve another `ListRepositoryTrackCommitsByRepositoryTrackNameRequest` endpoint that works upon `repository_name` and `repository_track_name` later (if ever).

What are your thoughts? This isn't necessarily hard blocking, but something we should consider before we move forward with the API.

Reply:
I guess you could say the same about the `Get` endpoint above (with names mixed with IDs). Maybe this isn't so bad. Let's look at the other APIs to make sure.